# 章節 11：假設檢定（Hypothesis Testing）

## 11.1 回到歐元問題

在 4.1 小節提到的問題從 David MacKay 的書籍 Information Theory, Inference, and Learning Algorithms 中寫到下述問題：

> 一個統計的論述出現在2002年1月4日星期五的英國《衛報》中：
旋轉一枚比利時的一歐元硬幣250次，出現140正面，110次背面。
一位倫敦經濟學院統計學講師 Barry Bligt 說到：「這令他非常懷疑。」
如果此硬幣是公平的，出現此結果的機率小於7%。
但這些證據是否指出此硬幣是不公平的嗎？

在第5章有提到，如果貝葉斯因子大於一的話，資料更傾向於假設而非另一個假設。

歐元問題的例子，我們考慮兩個假設：用 F 代表硬幣是公平的假設，B 代表硬幣是偏差的假設。

若硬幣是公平的，就很容易計算資料的似然性 p(D|F)。事實之前就已經完成此函數：

In [15]:
def Likelihood(data, hypo):
    x = hypo / 100.0
    heads, tails = data
    like = x**heads * (1-x)**tails
    return like

# 正面 140 次，反面 110 次
data = (140, 110)
f = Likelihood(data, 50)
print("(F)假設正面機率為 50% 看到此 data 的似然性：", f)

(F)假設正面機率為 50% 看到此 data 的似然性： 5.527147875260445e-76


上面的似然性只告訴我們說看到此資料的機率很小。用兩個似然性產生比率，所以我們還需要計算 p(D|B)。

但要如何計算 p(D|B)呢？因為「偏差」的意義不是這麼好定義。

一種可能是在定義「偏差」的假設前，先偷看資料。例如歐元問題我們可以說「偏差」的假設是出現正面的機率為 140/250。

In [18]:
actual_percent = 100.0 * 140 / 250
b_cheat = Likelihood(data, actual_percent)
print("(B)假設正面機率為 56% 看到此 data 的似然性：", b_cheat)

print("P(D|B)/P(D|F) 比率為：", b_cheat / f )

(B)假設正面機率為 56% 看到此 data 的似然性： 3.358289985239045e-75
P(D|B)/P(D|F) 比率為： 6.075990838368512


上方稱作 b_cheat；b_cheat 的似然性約為 3.4 * 10^-76。兩者的比例約為 6.1。

所以此資料是比較支持硬幣是偏差的假設 B。但用資料來制定假設是虛假的。

根據定義，除非觀察到的資料正面比例剛好 50%，否則任何的數據都會呈現支持假設 B 。